### PR TITLE
feat: Add support for ipfs addresses.

### LINKED ISSUE #15: Support for ipfs multiaddrs
We need to add a way to parse ipfs addrs to this module.

### PR DESCRIPTION
**Pull request by dignifiedquire**
Closes #15 

cc @diasdavid

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -389,7 +418,7 @@ describe('helpers', () => {
       )
 
       expect(
-        multiaddr('/http/0.0.0.0/utp').isThinWaistAddress()
+        multiaddr('/http/utp').isThinWaistAddress()
       ).to.be.eql(
```
Comment (diasdavid) [inline test/index.spec.js]: `eql?`

Comment (dignifiedquire) [inline test/index.spec.js]: eql === deep.equal

File: test/index.spec.js [hunk 2]
```diff
@@ -207,6 +221,21 @@ describe('variants', () => {
     expect(addr).to.have.property('buffer')
     expect(addr.toString()).to.equal(str)
   })
+
+  it('ip6 + tcp + websockets + ipfs', () => {
+    const str = '/ip6/2001:8a0:7ac5:4201:3ac9:86ff:fe31:7095/tcp/8000/websockets/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC'
```
Comment (diasdavid) [inline test/index.spec.js]: Can we have a test where we 'decapsulate' the ipfs part of a multiaddr like this and another test where we 'decapsulate' the websockets part, while checking the rest remains?

Comment (dignifiedquire) [inline test/index.spec.js]: @diasdavid Added a new test, let me know if that matches what you want

Comment (diasdavid) [inline test/index.spec.js]: thank you :)

**Pull request merged**